Use config file for config and other fixes

mfojtik opened the pull request:
@stevekuznetsov this refactors the code a little so we can use config map to pass the config and the
names of jobs we feed snowstorm (so we can update/disable/etc. them without recompiling)...

Also added option to remove "old" flakes, IOW. the flakes that we haven't seen for N days (if the last occurrence of this flake was 5-10 days ago, it is likely not a flake anymore)...

For testing this locally it will help to grant anonymous access to the GCS bucket. I added `.WithoutAuthentication()` option to GCS client, but without keys it complains with:

`
E0201 14:52:13.146238   15735 main.go:355] unable to get tests for name: could not get GCS bucket name: googleapi: Error 401: Anonymous caller does not have storage.buckets.get access to origin-ci-test., required
`

mfojtik committed:
Use config file for config and other fixes

stevekuznetsov commented on cmd/snowstorm/main.go:
```diff
@@ -485,46 +366,42 @@ func getFlakesForJob(name string, depth int, gcsBucket *storage.BucketHandle) er
-	if workersFlag != nil {
-		numWorkers = int(*workersFlag)
+	var err error
+	config, err = types.ParseConfig(*configPath)
```
Don't you need this to poll or `inotify` for hot reload?

mfojtik replied:
in ideal world, yes :)

stevekuznetsov commented on cmd/snowstorm/main.go:
```diff
@@ -18,40 +18,14 @@ import (
 	"cloud.google.com/go/storage"
 	"github.com/PuerkitoBio/goquery"
 	"github.com/golang/glog"
+	"github.com/mfojtik/snowstorm/cmd/snowstorm/types"
```
Import ordering wrong :D

stevekuznetsov commented on cmd/snowstorm/main.go:
```diff
@@ -485,46 +366,42 @@ func getFlakesForJob(name string, depth int, gcsBucket *storage.BucketHandle) er
 
 	ctx := context.Background()
-	gcsClient, err := storage.NewClient(ctx)
+	gcsClient, err := storage.NewClient(ctx, option.WithoutAuthentication())
```
Not sure we want this

mfojtik replied:
@stevekuznetsov this will guarantee you won't accidentally pass your local token

mfojtik commented:
@stevekuznetsov i updated the deploy.yaml to accommodate the config map (so config does not require rebuilding image/etc...)

also removed some of your logging as it was reaaaly lousy ;-)

in config you can specify the max days after which the flake won't show up on page anymore (so we can filter really old flakes).

stevekuznetsov commented:
This LGTM. Next we should try to get this to filter by job!
